chore: add new apex experiments surfaces types (#586)
discord-userdoccers/discord-userdoccers: pages/topics/experiments.mdx

</Alert>

> 

Reviewer: ```suggestion
```

| surface | integer | The [surface](#apex-experiment-surface) to return apex experiments for (only `APP`, `DEVELOPER_PORTAL`, `SEO`, and `MARKETING` is allowed) |

###### Apex Experiment Surface

| Value | Name             | Description                                                         |
| ----- | ---------------- | --------------------------------------------------------------------|
| 1     | API              | Return apex experiments that alter API functionality                |
| 2     | APP              | Return apex experiments that alter client functionality             |
| 3     | DEVELOPER_PORTAL | Return apex experiments that alter developer portal functionality   |
| 4     | ADMIN_PANEL      | Return apex experiments that alter admin panel functionality        |
| 5     | ADS_BUDGET_AB    | Return apex experiments that alter ads manager functionality        |
| 6     | AV_WORKER        | Return apex experiments that alter A/V worker functionality          |
| 7     | SEO              | Return apex experiments that alter SEO page functionality                |
| 8     | MARKETING        | Return apex experiments that alter marketing website functionality  |

> 

Reviewer: ```suggestion
```

<RouteHeader method="GET" url="/apex/seo-experiments/seo_url_slug/{slug}" unauthenticated>
  Get Apex Experiment Assignments for an invite slug
</RouteHeader>

> Returns an [apex experiments](#apex-experiments) object for the requesting invite slug.

Reviewer 2: ```suggestion
<RouteHeader method="GET" url="/apex/seo-experiments/seo_url_slug/{key}" unauthenticated>
  Get Apex Experiment Assignments For Slug
</RouteHeader>

Returns an [apex experiments](#apex-experiments) object for the given slug.
```
The parameter is called "key" in the api and has a maximum of 256 characters

<Alert type="info">

> The assignment object uses `seo_url_slug:{slug}` as identifiers for resources instead of snowflakes.

Reviewer 2: ```suggestion
The returned assignment will have [`CUSTOM`](#apex-experiment-unit-type) type and `seo_url_slug:<key>` as the unit ID.
```

I think the type in assignments structure should be changed to `any` and the id type should be in unit type table in separate column (probably "Unit ID Type" or similar)

Author: I think it should be `string` and in the description we specify the actual type for  seo slug endpoint etc

Author: After a bit of research, I figured out that specific experiments can have a `custom_unit_prefix`, in this case it is `seo_url_slug:`

| Field   | Type    | Description                                                                                                           |
| ------- | ------- | --------------------------------------------------------------------------------------------------------------------- |

> | surface | integer | The [surface](#apex-experiment-surface) to return apex experiments for (only `APP` and `DEVELOPER_PORTAL` and `SEO` and `MARKETING` is allowed) |

Reviewer: ```suggestion
| surface | integer | The [surface](#apex-experiment-surface) to return apex experiments for (only `APP`, `DEVELOPER_PORTAL`, `SEO`, and `MARKETING` is allowed) |
```

###### Apex Experiment Surface

| Value | Name             | Description                                                         |
| ----- | ---------------- | --------------------------------------------------------------------|
| 1     | API              | Return apex experiments that alter API functionality                |
| 2     | APP              | Return apex experiments that alter client functionality             |
| 3     | DEVELOPER_PORTAL | Return apex experiments that alter developer portal functionality   |
| 4     | ADMIN_PANEL      | Return apex experiments that alter admin panel functionality        |
| 5     | ADS_BUDGET_AB    | Return apex experiments that alter ads manager functionality        |

> | 6     | AV_WORKER        | Return apex experiments that alter av worker functionality          |

Reviewer: ```suggestion
| 6     | AV_WORKER        | Return apex experiments that alter A/V worker functionality          |
```

> | 7     | SEO              | Return apex experiments that alter seo functionality                |

Reviewer: ```suggestion
| 7     | SEO              | Return apex experiments that alter SEO page functionality                |
```
